conans/client/conan_api.py

```diff
@@ -0,0 +1,629 @@
+        current_path = os.path.abspath(path or os.getcwd())
+        self._manager.export(user, current_path, keep_source, filename=filename)
+
+    @api_method_decorator
+    def remove(self, pattern, query=None, packages=None, builds=None, src=False, force=False, remote=None):
+        reference = _check_query_parameter_and_get_reference(query, pattern)
+
+        if packages is not None and query:
+            raise ConanException("'-q' and '-p' parameters can't be used at the same time")
+
+        if builds is not None and query:
+            raise ConanException("'-q' and '-b' parameters can't be used at the same time")
+
+        self._manager.remove(reference or pattern, package_ids_filter=packages, build_ids=builds,
+                             src=src, force=force, remote=remote, packages_query=query)
+
+    @api_method_decorator
+    def copy(self, reference="", user_channel="", force=False, all=False, package=None):
+        reference = ConanFileReference.loads(reference)
+        new_ref = ConanFileReference.loads("%s/%s@%s" % (reference.name,
+                                                         reference.version,
+                                                         user_channel))
+        if all:
+            package = []
+        self._manager.copy(reference, package, new_ref.user, new_ref.channel, force)
+
+    @api_method_decorator
+    def user(self, name=None, clean=False, remote=None, password=None):
+        if clean:
+            localdb = LocalDB(self._client_cache.localdb)
+            localdb.init(clean=True)
+            self._user_io.out.success("Deleted user data")
+            return
+        self._manager.user(remote, name, password)
+
+    @api_method_decorator
+    def search(self, pattern, query=None, remote=None, case_sensitive=False):
+        reference = _check_query_parameter_and_get_reference(query, pattern)
+
+        self._manager.search(reference or pattern,
+                             remote,
+                             ignorecase=not case_sensitive,
+                             packages_query=query)
+
+    @api_method_decorator
+    def upload(self, pattern, package=None, remote=None, all=False, force=False, confirm=False, retry=2, retry_wait=5,
+               skip_upload=False):
+        """ Uploads a package recipe and the generated binary packages to a specified remote
+        """
+        if package and not is_a_reference(pattern):
+            raise ConanException("-p parameter only allowed with a valid recipe reference, not with a pattern")
+
+        self._manager.upload(pattern, package,
+                             remote, all_packages=all,
+                             force=force, confirm=confirm, retry=retry,
+                             retry_wait=retry_wait, skip_upload=skip_upload)
+
+    @api_method_decorator
+    def remote(self, subcommand, reference=None, verify_ssl=False, remote=None, url=None):
```

Comment:
Same as above, an api with the commands as parameters seem a bit narrow-api. But just an opinion, no strong view.